### PR TITLE
timer: only run when attached (SC-1304)

### PR DESCRIPTION
**Pull request by orndorffgrant**
no-lp no-gh
The final PR of my 23.04 tech-debt ticket.

After our recent improvements, we only need an intermittent job to run when attached. So let's not waste everyone's cpu cycles by leaving the timer on when unattached.

## Test Steps

Verify the timer is off when upgrading to this version while unattached.
Verify the timer stays on when upgrading to this version while attached.
Verify that attaching starts the timer.
Verify that detaching stops the timer.

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [x] Yes
 - [ ] No

### CODE REVIEW

File: uaclient/timer/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import logging
+from subprocess import TimeoutExpired
+
+from uaclient import exceptions, system
+
+LOG = logging.getLogger("uaclient.timer")
```
Comment (aciba90) [inline uaclient/timer/__init__.py:6]: Portability nit:

```suggestion
LOG = logging.getLogger(__package__)
```

**Review by aciba90 — APPROVED** [review]
LGTM, thanks!

One question. What is the reason to not start the timer right after a successful auto-attach? I understand it will eventually get activated at some point per `systemd/ua-timer.timer`, but I do not see the reason why in the case of attach with token a timer.start() is needed but not ofr the auto-attach case.

File: uaclient/timer/__init__.py (hunk 1 above)
Comment (orndorffgrant) [reply uaclient/timer/__init__.py:6]: Good point! Is there a reason to prefer `__package__` over `__name__`?

**Comment by orndorffgrant** [comment]
@aciba90 Thanks for taking a look! auto-attach includes calling `attach_with_token`, so this should start the timer right after an auto-attach as well

**Comment by renanrodrigo** [comment]
All succeeded all approved clicking the magic button

**Pull request merged**